docs/coding_guidelines.md

```diff
@@ -0,0 +1,84 @@
+# Coding guidelines for OCS-CI project
+
+Lets follow the following guidelines for the Core Libraries and Test Scripts
+
+* Follow pep8 or use autopep8 to automagically conform to pep8 style.
+* Each Python Class is well documented for inheritance, methods and
+    attributes. Add sufficient comments for code block that is easy
+    to understand for others to review.
+    Use [Google Style of Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
+* Code Readability takes priority, Lets avoid crazy hacks and tricks
+    and stick to straight forward code blocks that are easy to
+    understand for all.
+* For core libraries when possible please add the unittest to
+    minimize regression for many tests
+* Helper functions that are useful to wide variety of tests should
+    reside in utils and/or should be carefully reviewed to check if
+    they can actually be methods.
+* test_scripts will reside in test/ folders and will define `run`
+    function
+* test_scripts should return non zero exit status for failure and 0
+    for 'pass'
+
+**ex: test_scripts/test_ocs_operator.py**
+
+```python
+        def run(ocs_context, **kw):
+            # use ocs_context to deal with cluster
+            # run your test steps
+            if test_failed:
+                return False
+            else:
+                return Pass
```

Comment:
@vasukulkarni I copied this from [here](https://github.com/red-hat-storage/ocs-ci/blob/master/docs/CodingGuidelinesandReview), but when I am looking to example the `Pass` variable value should be 0 here as I understand?

Or do we also allow just `pass` or not returning any value here (default return value is None).
Maybe worth to update this doc to not confuse here with `Pass` as Python keyword is `pass`.

Reply:
thanks for pointing out, Yeah I think its better to use 0 and 1 there instead of False and Pass, once we update the enum I can fix that.

Reply:
@vasukulkarni ok, I fixed this one in commit https://github.com/red-hat-storage/ocs-ci/pull/7/commits/db5b515eb0673d45de15d6102086024646511aed so please approve change again, I hope I will get tomorrow some other approve from @RazTamir @dahorak @clacroix12 and I will merge this one. Thanks

Reply:
+1 on that.
As soon as we'll have the ENUM, let's have 'test_pass=0' and 'test_fail=1'